src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/Install/SetupXPackPasswordsTask.cs

```diff
@@ -111,5 +119,63 @@ private void SetPassword(HttpClient client, string elasticUserPassword, string u
 			}
 			this.Session.SendProgress(200, $"Changed password for user '{user}'");
 		}
+
+		// Might be moved and refactored when is needed for other tasks
+		public static string GetBaseAddress(string networkHost, int? httpPort)
+		{
+			var port = httpPort ?? DefaultHttpPort;
+
+			if (string.IsNullOrEmpty(networkHost))
+				return $"http://localhost:{port}/";
+
+			var host = networkHost;
+			if (networkHost.IndexOf(',') >= 0) // Uri host must not contain comma, but elasticsearch.yml supports it
+			{
+				var hosts = networkHost.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries).Select(h => h.Trim());
+				host = hosts.FirstOrDefault(h => h == "localhost" || h == "127.0.0.1"
+					|| h == LocalhostPlaceholder || h == LocalhostV4Placeholder || h == LocalhostV6Placeholder) // prefer local interface
+					?? hosts.First();
+			}
+
+			switch (host)
```

Comment:
This is great, it attempts to respect the special placeholder syntax from elasticsearch when bootstrapping http calls.